txnkv/transaction/txn.go

```diff
@@ -1185,10 +1199,15 @@ func (txn *KVTxn) lockKeys(ctx context.Context, lockCtx *tikv.LockCtx, fn func()
 		}
 
 		if txn.IsInAggressiveLockingMode() {
+			actualForUpdateTS := lockCtx.ForUpdateTS
+			if val.LockedWithConflictTS > actualForUpdateTS {
```

Comment:
Should we add check or assert `if val.LockedWithConflictTS > 0 then val.LockedWithConflictTS > lockCtx.ForUpdateTS` to ensure no unexpected cases would happen.

Reply:
Considering that the content of the response depends on TiKV, I think returning error is better than panicking; then considering that the loop updates keys' states to membuffer, I think it's better not to break it and return on the halfway. So I checked the error after this loop. PTAL again.